app/src/main/java/org/mozilla/tv/firefox/telemetry/TelemetryFactory.kt

```diff
@@ -39,8 +38,12 @@ object TelemetryFactory {
                         TelemetrySettingsProvider.APP_ID
                 )
                 .setSettingsProvider(TelemetrySettingsProvider(context))
-                .setCollectionEnabled(telemetryEnabled)
-                .setUploadEnabled(telemetryEnabled)
+
+        context.serviceLocator.settingsRepo.dataCollectionEnabled.observeForever { collectionEnabled ->
+            if (collectionEnabled != null) {
+                configuration.setUploadEnabled(collectionEnabled).isCollectionEnabled = collectionEnabled
```

Comment:
Optional nit: this might just be me, but combining the set and property syntax here briefly confused me.  Suggest breaking this into two lines of property syntax.